File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {

        if (head == null || head.next == null || k == 0) return head;

        ListNode temp=head;
        ListNode x=head;
        int count=1;

        while(temp.next!=null){
            count++;
            temp=temp.next;
        }
temp.next=head;
        int p=k%count;
        int step=count-p;
    ListNode newtail=head;
        for(int i=1;i<step;i++){
newtail=newtail.next;
        }

        ListNode newhead=newtail.next;
        newtail.next=null;

        return newhead;
        
    }
}

File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {

if(head==NULL){
    return NULL;
}

        int count=0;

        vector<int>nums;

        ListNode* temp=head;
        while(temp!=NULL){
            nums.push_back(temp->val);
            count++;
            temp=temp->next;
        }
        k=k%count;

        reverse(nums.begin(),nums.end());
        reverse(nums.begin(),nums.begin()+k);
        reverse(nums.begin()+k,nums.end());

        ListNode* x=new ListNode(nums[0]);
ListNode* curr=x;
        int i=1;
        int n=nums.size();
        while(i<n){
            ListNode* p=new ListNode(nums[i]);
     curr->next=p;
     curr=curr->next;
        //   curr=curr->next;
            i++;
        }
        return x;
    }
};